assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
is this super important, it will still compile error if you mess up the types right? The generated code would be incompatible with your written constructor.